### DISCUSSION TITLE
how do I get upstream host in serverless-pre-function?

### DISCUSSION THREAD

**Discussion by dark-code** [apache/apisix#8973] [Q&A]
1. how do I get upstream host in serverless-pre-function?
2. where is the api doc?

**Comment by tokers** [comment]
> how do I get upstream host in serverless-pre-function?

For getting the upstream host, you may need to read codes in `apisix/upstream.lua` and check how APISIX calculates the upstream host.

> where is the api doc?

Which API? We have the API spec for Admin API and Control API.

**Reply by dark-code** [reply]
api doc means :when we develope a plugin（or write a pre function）  which function I can use and how to use 。just like javadoc

**Reply by dark-code** [reply]
I already check many times `apisix/upstream.lua` but don't find any function can use to get upstream or upstream's node host info

**Reply by rohitkrishna-marneni** [reply]
Were you able to figure this out?
